service/controller/chart/v1/resource/tiller/resource.go

```diff
@@ -56,7 +57,12 @@ func (r *Resource) ensureTillerInstalled(ctx context.Context) error {
 		"spec.template.spec.tolerations[0].operator=Exists",
 	}
 	err := r.helmClient.EnsureTillerInstalledWithValues(ctx, values)
-	if err != nil {
+	if helmclient.IsTooManyResults(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "currently too many tiller pods")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		reconciliationcanceledcontext.SetCanceled(ctx)
+		return nil
```

Comment:
During upgrades there can be multiple pods.

chart-operator can't do anything without a tiller pod. So it's safer to cancel the reconciliation and wait for the tiller deployment.

Reply:
This seems like a dangerous situation to me. 
Failure case: There is another tiller running in the ns but no upgrade is happening (e.g. customer installed it).

There is no way to detect this situation anymore or alert on it. Is there a too many tillers metric? Can we add one and alert if there is more than 1 tiller in the namespace for e.g. 1 hour?

Reply:
Good point. I'll add that metric to the collector here since chart-operator is deployed everywhere.

Of course I'd rather invest the time in the Helm 3 upgrade but we are where we are.

Reply:
I agree with the helm 3 point but I can't tell how far away we are from that / how much effort that's going to be.

Reply:
We're doing the [Helm 2.16.1 upgrade](https://github.com/giantswarm/giantswarm/issues/7675) first for K8s 1.16. We can then plan the [Helm 3 upgrade](https://github.com/giantswarm/giantswarm/issues/7676).

I created issues yesterday but no ETA on Helm 3 yet.